packages/flutter/lib/src/widgets/async.dart

```diff
@@ -596,6 +596,8 @@ class _FutureBuilderState<T> extends State<FutureBuilder<T>> {
   /// or after widget reconfiguration to a new Future.
   Object? _activeCallbackIdentity;
   late AsyncSnapshot<T> _snapshot;
+  Future<T>? _future;
+  Widget? _lastWidget;
```

Comment:
You should not caches widget in the state, that is what didUpdateWidget is for.

Can you apply the fix to didUpdateWidget instead?